pkg/kubelet/certificate/certificate_manager.go

```diff
@@ -76,55 +124,44 @@ type manager struct {
 	certAccessLock           sync.RWMutex
 	cert                     *tls.Certificate
 	shouldRotatePercent      uint
+	forceRotation            bool
 }
 
 // NewManager returns a new certificate manager. A certificate manager is
 // responsible for being the authoritative source of certificates in the
 // Kubelet and handling updates due to rotation.
-func NewManager(
-	certSigningRequestClient certificatesclient.CertificateSigningRequestInterface,
-	template *x509.CertificateRequest,
-	usages []certificates.KeyUsage,
-	certificateStore Store,
-	certRotationPercent uint) (Manager, error) {
-
-	cert, err := certificateStore.Current()
+func NewManager(config *Config) (Manager, error) {
+	cert, forceRotation, err := getCurrentCertificateOrBootstrap(
+		config.CertificateStore,
+		config.BootstrapCertificatePEM,
+		config.BootstrapKeyPEM)
 	if err != nil {
 		return nil, err
 	}
 
-	if certRotationPercent > 100 {
-		certRotationPercent = 100
+	if config.CertificateRotationPercent > 100 {
+		return nil, fmt.Errorf("certificate rotation percent was over 100%%")
 	}
 
 	m := manager{
-		certSigningRequestClient: certSigningRequestClient,
-		template:                 template,
-		usages:                   usages,
-		certStore:                certificateStore,
+		certSigningRequestClient: config.CertificateSigningRequestClient,
+		template:                 config.Template,
+		usages:                   config.Usages,
+		certStore:                config.CertificateStore,
 		cert:                     cert,
-		shouldRotatePercent:      certRotationPercent,
+		shouldRotatePercent:      config.CertificateRotationPercent,
+		forceRotation:            forceRotation,
```

Comment:
We talked about changing the sync loop to just wait until the certificate needs to be rotated. If you take that approach, you could replace forceRotation with setting the rotation to be immediate. Doesn't need to happen in this PR, but just mentioning the idea here...